### DISCUSSION TITLE
Unrecognized x509 Certificate Error

### DISCUSSION THREAD

**Discussion by J-Hunter-Hawke** [smallstep/certificates#397] [Q&A] [answered]
Hey guys,

I have my intermediate certificate authority configured and running in an AWS account, and I am able to successfully bootstrap off of it as seen below:
![image](https://user-images.githubusercontent.com/26150306/96027654-2d826b00-0e26-11eb-90c5-ae8c96bee361.png)

And inspecting the certificate shows that I have downloaded the correct file. However, any subsequent calls to the certificate authority fail. For example:
![image](https://user-images.githubusercontent.com/26150306/96027862-87833080-0e26-11eb-9a47-ec21578d7f4f.png)
![image](https://user-images.githubusercontent.com/26150306/96027960-b1d4ee00-0e26-11eb-85da-f1bec7a2ae8b.png)

The certificate authority is sitting behind an application load balancer in AWS which is using a managed certificate from Amazon Certificate Manager. From the latter error, it looks like the step cli doesn't trust Amazon's certificate, so it is refusing to connect.

When I check the certificate authority logs, I do not see my request. Additionally, I'm not even seeing the request come through in the application load balancer logs, so it looks like no connection is being established from my computer to the CA.

Has anyone troubleshooted such an error before or have any suggestions on how to work around this? It seems like I should use the root CA to create a custom x509 certificate for the ALB, but I would like to avoid this route if possible.

Thanks in advance!
-Hunter

**Comment by tashian** [comment, marked as answer]
Hi @J-Hunter-Hawke,

Good question. I think you're not even making it through the TLS handshake with the load balancer, and that's why nothing is showing in any of your request logs. `step` expects to be able to establish a mutual TLS connection with the CA. Application load balancing would require your ALB to decrypt and encrypt messages and essentially impersonate the CA, and we don't recommend that from a security standpoint. It's also a pain to configure.

Best to have the CA and client speak directly to each other or via network load balancing or TLS passthrough.

Best,
Carl

**Comment by maraino** [comment]
@J-Hunter-Hawke `step` by default is configured to only trust one root certificate, generally the root certificate in step-ca.
If you want to support a different one, the best option is to append the load balancer root certificate to `$(step path)/certs/root_ca.crt`.

**Comment by J-Hunter-Hawke** [comment]
Thank you @tashian and @maraino for the suggestions. I have switched over to using an NLB without an ACM certificate, and the certificate authority is working as expected. However, I will keep note that we can append a second certificate if need be in the future.

**Reply by maraino** [reply]
That's a better option, with the TLS termination in the LB there are some operations like mTLS renewal that are not going to work.
